File: 2020 czerwiec/python/1.py
with open("pary.txt", "r") as f:
    pary = f.read().strip().split('\n')

def is_prime(n):
    for i in range(2, n):
        if n % i == 0:
            return False
    return True

for p in pary:
    n, s = p.split()
    n = int(n)

    if n % 2 == 0:
        i = 3
        while not is_prime(i) or not is_prime(n - i):
            i += 1
        print(n, i, n - i)

File: 2020 czerwiec/c++/1.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

bool is_prime(int n)
{
    for (int i = 2; i * i <= n; i++)
    {
        if (n % i == 0)
            return false;
    }
    return true;
}

int main()
{
    ifstream in("pary.txt");

    int a;
    string s;
    while (in >> a >> s)
    {
        if (a % 2 == 0)
        {
            int i = 3;
            while (!is_prime(i) || !is_prime(a - i))
                i++;
            cout << a << ' ' << i << ' ' << a - i << '\n';
        }
    }
}